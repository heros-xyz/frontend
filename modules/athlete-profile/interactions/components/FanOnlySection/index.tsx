import {
  Button,
  Flex,
  Modal,
  ModalContent,
  ModalOverlay,
  Text,
  useDisclosure,
} from "@chakra-ui/react";
import ExclamationIcon from "@/components/svg/Exclamation";
import { LockCloseIcon } from "@/components/svg/Settings";

const FanOnlySection = () => {
  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <>
      <Modal onClose={onClose} isOpen={isOpen} isCentered>
        <ModalOverlay />
        <ModalContent
          pt="20px"
          pb={{ base: "26px", lg: "30px" }}
          maxW="unset"
          w={{ base: "344px", lg: "740px" }}
        >
          <Flex
            flexDirection="column"
            justifyContent="center"
            alignItems="center"
            textAlign="center"
          >
            <ExclamationIcon
              w={{ base: "48px", lg: "80px" }}
              h={{ base: "48px", lg: "80px" }}
              mb={{ base: "15px", lg: "20px" }}
            />
            <Text
              fontWeight="medium"
              fontSize={{ base: "lg", lg: "2xl" }}
              mb="5px"
            >
              Restricted to fans only
            </Text>
            <Text
              fontWeight="normal"
              fontSize={{ base: "base", lg: "xl" }}
              mb={{ base: "20px", lg: "30px" }}
            >
              Please subscribe to view all content from your favorite athlete.
            </Text>
            <Button
              size="lg"
              variant="primary"
              w={{ base: "312px", lg: "137px" }}
              mb={{ base: "15px", lg: "25px" }}
            >
              JOIN NOW
            </Button>
            <Button
              onClick={onClose}
              textDecoration="underline"
              variant="link"
              textTransform="unset"
            >
              Cancel
            </Button>
          </Flex>
        </ModalContent>
      </Modal>

      <Button
        size="lg"
        onClick={onOpen}
        bg="secondary"
        gap="10px"
        color="black"
        w="full"
      >
        <Text> FANS ONLY </Text>
        <LockCloseIcon />
      </Button>
    </>
  );
};

export default FanOnlySection;
