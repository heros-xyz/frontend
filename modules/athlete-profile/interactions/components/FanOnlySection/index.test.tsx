import { render } from "@testing-library/react";
import FanOnlySection from ".";

it("renders Fan only section unchanged", () => {
  const { container } = render(<FanOnlySection />);
  expect(container).toMatchSnapshot();
});
