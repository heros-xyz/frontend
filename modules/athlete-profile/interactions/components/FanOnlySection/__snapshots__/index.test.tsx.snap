// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Fan only section unchanged 1`] = `
<div>
  <button
    class="chakra-button css-96i9t"
    type="button"
  >
    <p
      class="chakra-text css-0"
    >
       FANS ONLY 
    </p>
    <svg
      class="chakra-icon css-nyt2fo"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 15V17M6 21H18C19.1046 21 20 20.1046 20 19V13C20 11.8954 19.1046 11 18 11H6C4.89543 11 4 11.8954 4 13V19C4 20.1046 4.89543 21 6 21ZM16 11V7C16 4.79086 14.2091 3 12 3C9.79086 3 8 4.79086 8 7V11H16Z"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="1.5"
      />
    </svg>
  </button>
</div>
`;
